pkg/apis/pipeline/v1alpha1/conversion_error.go

```diff
@@ -26,6 +26,10 @@ const (
 	// resources when they cannot be converted to warn of a forthcoming
 	// breakage.
 	ConditionTypeConvertible apis.ConditionType = v1beta1.ConditionTypeConvertible
+	// Conversion Error Field for finally
+	ConversionErrorFinally = "Finally"
+	// Conversion Error descriptive message for finally
+	ConversionErrorFinallyMsg = ConversionErrorFinally + " is not available in v1alpha1"
```

Comment:
yup we certainly need some docs or set of examples to guide us through the conversion process.

Adding `finally` specific message here to be very specific in case of conversion failures. There is only one today, not sure where are we headed with conversion process, @vdemeester need your help here to include field specific or generic message?

Reply:
I would think we could go for a generic one (and have the code "add" the name of the missing field)

Reply:
@vdemeester @bobcatfish I have added a generic message here:

```
// Conversion Error message for a field not available in v1alpha1
ConversionErrorFieldNotAvailableMsg = "the specified field/section is not available in v1alpha1"
```

And, field name is added from the conversion function:

```
const FinallyFieldName = "finally"

// finally clause was introduced in v1beta1 and not available in v1alpha1
if len(source.Finally) > 0 {
	return ConvertErrorf(FinallyFieldName, ConversionErrorFieldNotAvailableMsg)
}
```